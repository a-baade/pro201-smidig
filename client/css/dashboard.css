.dashboard-page{
    display: flex;
    flex-direction: row;
    width: 100%;
    background-color: #e5e5e5;
}

/*Code for sidebar*/
.dashboard-sidebar{
    display: flex;
    flex-direction: column;
    background-color: #35656f;
    padding: 2em;
    color: #ffffff;
}

.dashboard-pfp-name{
    display: inherit;
    flex-direction: column;
    align-items: center;
}

.dashboard-pfp{
    border-radius: 100%;
    width: 7em;
}

/*Code for main*/
.dashboard-main{
    display: flex;
    flex-direction: column;
    width: 100%;
    max-width: 1200px;
    margin-left: auto;
    margin-right: auto;
    padding: 4em;
}

.chart-wrap{
    display: flex;
    flex-direction: row;
    justify-content: center;
}

/*Code for top row (boxes with information)*/
.dashboard-top-row{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
}

.dashboard-info-box{
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    width: 10em;
    height: 5em;
    border-radius: 1em;
    font-size: 25px;
    background-color: #8ab7ba;
}

/*Code for second row (Chart with donations over time)*/
.dashboard-row-2{
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0 1em 1em 1em;
    margin-top: 1em;

    background-color: #ffffff;
    border-radius: 1em;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
}

.chart2-sidebar{
    display: flex;
    flex-direction: column;
    text-align: center;
    margin-left: 0em;
    margin-top: 4em;
}

#months{
    -moz-appearance: none;
    -webkit-appearance: none;
    appearance: none;

    height: 3em;

    background-color: #c0cfc8;
    border: none;
}

.chart2-circle{
    display: flex;
    justify-content: center;
    align-items: center;
    align-text: center;
    margin-top: 1em;
    background-color: #fbe8da;
    border-radius: 100%;
    width: 11em;
    height: 11em;
}

.chart2{
    margin-right: 3em;
}

/*Code for third row (Donation chart compared to other organizations)*/
.dashboard-row-3{
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0 1em 1em 1em;
    margin-top: 1em;

    background-color: #66a2ad;
    color: #ffffff;
    border-radius: 1em;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
}

.chart1-sidebar{
    margin-top: 3em;
}

.chart1-circle{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    background-color: #ffffff;
    color: #35656f;
    border: 0.5em solid #35656f;
    border-radius: 100%;
    width: 10em;
    height: 10em;
}

.percent-txt{
    font-weight: bold;
    font-size: 35px;
    margin-bottom: 0;
    margin-top: 0;
}

.chart1-circle-txt{
    margin-top: 0;
    border-top: 2px solid #35656f;
}

.chart1-textbox{
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    margin-top: 1em;
    width: 11em;
    height: 3em;

    background-color: #35656f;
    border-radius: 0.5em;
}

.chart1{
    margin-right: 3em;
}







