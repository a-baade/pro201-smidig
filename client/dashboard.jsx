import React from "react";

export function Dashboard() {
  return <h1>Dashboard</h1>;
}