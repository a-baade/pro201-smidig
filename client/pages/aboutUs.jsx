import React from "react";
import MainBg from "../pics/about-us-pictures/main-bg.png";
import FirstLayerArt from "../pics/about-us-pictures/firstLayerArt.png";
import TopLayerArt from "../pics/about-us-pictures/topLayerArt.png";
import PovertyFirst from "../pics/about-us-pictures/povertyFirstLayer.png";
import PovertySecond from "../pics/about-us-pictures/povertytopLayer.png";
import ZeroHunger from "../pics/about-us-pictures/zeroHungerFirst.png";
import ZeroHungerTop from "../pics/about-us-pictures/seroHungerTop.png";
import GoodHealthFirst from "../pics/about-us-pictures/goodHealthFirst.png";
import GoodHealthTop from "../pics/about-us-pictures/goodHealthTop.png";
import QualityEducation from "../pics/about-us-pictures/qualityEducation.png";
import QualityEducationTop from "../pics/about-us-pictures/qualityEducationTop.png";
import GenderEquality from "../pics/about-us-pictures/genderEqualityFirst.png";
import GenderEqualityTop from "../pics/about-us-pictures/genderEqualityTop.png";

export function AboutUs() {
  return (
    <div>
      <div>
        <img src={MainBg} className={"main-bg-pic"} alt={"mainBg"}/>
      </div>
      <div className={"second-row-box"}>
        <div className={"knowledge-text-box"}>
          <p className={"our-mission"}>OUR MISSION</p>
          <h1 className={"knowledge-power"}>KNOWLEDGE = POWER</h1>
          <p>We're a Norwegian female-founded startup changing corporate giving and the way companies and nonprofits connect.</p>
          <p>We're saving companies resources, help create stable income streams for nonprofits, and create deeper relationships between donor and nonprofit.</p>
          <p> We help implement solid strategies for impactful and sustainable giving, all in line with the UN SDGs.</p>
        </div>
        <div className={"art-box"}>
          <img src={FirstLayerArt} className={"first-layer-art"} alt={"firstLayer"}/>
          <img src={TopLayerArt} className={"top-layer-art"} alt={"topLayer"}/>
          <p className={"our-mission mission"}>OUR MISSION</p>
          <h1 className={"action-call"}>A CALL TO ACTION</h1>
        </div>
      </div>
      <div className={"big-container"}>
        <div className={"small-box light"}>
          <p className={"small-title"}>No poverty</p>
          <p className={"small-txt"}>We are committed to end poverty in all its forms everywhere</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Zero hunger</p>
          <p className={"small-txt"}>End hunger, achieve food security and improve nutrition and promote sustainable agriculture</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <img src={GoodHealthFirst} className={"good-health"}/>
          <img src={GoodHealthTop} className={"good-health-top"}/>
          <p className={"small-title good-health-title"}>Good health</p>
          <p className={"small-txt good-health-txt"}>Ensure healthy lives and promote well-being for all at all ages</p>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Quality education</p>
          <p className={"small-txt"}>Ensure inclusive and equitable quality education and promote lifelong learning opportunities for all</p>
          <img src={QualityEducation} className={"quality-education"}/>
          <img src={QualityEducationTop} className={"quality-education-top"}/>
        </div>
        <div className={"small-box dark"}>
          <img src={GenderEquality} className={"gender-equality"}/>
          <img src={GenderEqualityTop} className={"gender-equality-top"}/>
          <p className={"small-title gender-eql-title"}>Gender equality</p>
          <p className={"small-txt gender-eql-txt"}>Achieve gender equality and empower all women and girls</p>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Clean water & sanitation</p>
          <p className={"small-txt"}>Ensure availability and sustainable management of water and sanitation for all</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Affordable & clean energy</p>
          <p className={"small-txt"}>Ensure access to affordable, reliable, sustainable and modern energy for all</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Decent work & economic growth</p>
          <p className={"small-txt"}>Promote sustained, inclusive and sustainable economic growth, full and productive employment and decent work for all</p>
          <img src={ZeroHunger} className={"decent-work"}/>
          <img src={ZeroHungerTop} className={"decent-work-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Industry, innovation & infrastructure</p>
          <p className={"small-txt"}>Build resilient infrastructure, promote inclusive and sustainable industrialization and foster innovation</p>
          <img src={PovertyFirst} className={"industry"}/>
          <img src={PovertySecond} className={"industry-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Reduce inequalities</p>
          <p className={"small-txt"}>Reduce inequality within and among countries</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Subsainable cities & communinites</p>
          <p className={"small-txt"}>Make cities and human settlements inclusive, safe, resilient and sustainable</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>responsible consumption & production</p>
          <p className={"small-txt"}>Ensure sustainable consumption and production patterns</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Climate action</p>
          <p className={"small-txt"}>Take urgent action to combat climate change and its impacts</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Life below water</p>
          <p className={"small-txt"}>Conserve and sustainably use the oceans, seas and marine resources for sustainable development</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Life on land</p>
          <p className={"small-txt"}>Protect, restore and promote sustainable use of terrestrial ecosystems, sustainably manage forests, combat desertification, and halt and reverse land degradation and halt biodiversity loss</p>
          <img src={PovertyFirst} className={"life-first"}/>
          <img src={PovertySecond} className={"life-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Peace, justice & strong institution</p>
          <p className={"small-txt"}>Promote peaceful and inclusive societies for sustainable development, provide access to justice for all and build effective, accountable and inclusive institutions at all levels</p>
          <img src={ZeroHunger} className={"peace-first"}/>
          <img src={ZeroHungerTop} className={"peace-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Partnerships for the goals</p>
          <p className={"small-txt"}>Strengthen the means of implementation and revitalize the global partnership for sustainable development</p>
          <img src={PovertyFirst} className={"partnership-first"}/>
          <img src={PovertySecond} className={"partnership-second"}/>
        </div>

        <div className={"small-box dark"}>
          <img src={ZeroHunger} className={"un-goals-first"}/>
          <img src={ZeroHungerTop} className={"un-goals-top"}/>
          <p className={"un-goals-title"}>THE UN SDG GOALS</p>
        </div>
      </div>
    </div>
  );
}