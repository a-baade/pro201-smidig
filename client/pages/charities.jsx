import React, { useContext, useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { ApiContext } from "../apiContext";
import { useLoading } from "../useLoading";
import { fetchJSON } from "../lib/fetchJSON";
import LoadingWheel from "../components/loadingSpinner";

function CharityCard({
  charity: { _id, name, description, bgImage, charityLogo },
}) {
  // Charity card with button to donate or charity page
  return (
    <div className={"charity-card custom-shadow"}>
      <div
        className={"charity-background"}
        style={{ backgroundImage: `url( data:image/png;base64${bgImage} )` }}
      >
        <img
          src={`data:image/png;base64${charityLogo}`}
          className={"charity-img custom-shadow"}
        />
      </div>

      <div className={"charity-card-content"}>
        <h3>{name}</h3>
        <p>{description}</p>
        <div className={"card-buttons"}>
          <Link to={{ pathname: "/charities/charity/id?id=" + _id }}>
            <button className="card-button custom-shadow-button">
              Read more
            </button>
          </Link>
          <Link to={{ pathname: "/charities/donate/id?id=" + _id }}>
            <button className="card-button custom-shadow-button">Donate</button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export function Charities() {
  const { allCharities } = useContext(ApiContext);
  const { category, setCategory } = useState("");
  const { categoryQuery, setCategoryQuery } = useState("");
  const [search, setSearch] = useState("");
  const [data, setData] = useState([]);
  const { loading, error } = useLoading(
    async () => await allCharities({ category }),
    [category]
  );

  useEffect(async () => {
    if (search === "") {
      await fetchJSON("/api/charities").then((jsonData) => {
        setData(jsonData);
      });
    } else {
      await fetchJSON(`/api/charities/search/?name=${search}`).then(
        (jsonData) => {
          setData(jsonData);
        }
      );
    }
  }, [search]);

  function handleSubmitQuery(e) {
    e.preventDefault();
    setCategory(categoryQuery);
  }

  async function handleSearch(event) {
    await setSearch(event);
  }

  if (loading) {
    return <LoadingWheel />;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  // Add category filter
  return (
    <div className={"page-margin"}>
      <div className="container-charities">
        <div className="search-charities">
          <input
            className="search-input-charities"
            placeholder="Search Charities"
            value={search}
            onChange={(e) => handleSearch(e.target.value)}
          />
          {search && (
            <div className={"charity-results"}>Results for {search}</div>
          )}
        </div>
        <div></div>

        {/*<form onSubmit={handleSubmitQuery}>*/}
        {/*    <button  id={"category-query"}*/}
        {/*             value={categoryQuery}*/}
        {/*             onChange={(e) => setCategoryQuery(e.target.value)}>Water</button>*/}
        {/*    <button  id={"category-query"}*/}
        {/*             value={categoryQuery}*/}
        {/*             onChange={(e) => setCategoryQuery(e.target.value)}>Knowledge</button>*/}
        {/*</form>*/}
      </div>
      <div>
        <ul className={"charity-cards-container"}>
          {data?.map((charity, index) => (
            <CharityCard key={index} charity={charity} />
          ))}
        </ul>
      </div>
    </div>
  );
}
