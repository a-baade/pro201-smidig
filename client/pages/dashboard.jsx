import React from "react";
import { Chart2 } from "../components/charts";
import { Chart1 } from "../components/charts";


export function Dashboard({ user }) {
  return (
      <div className={"dashboard-page"}>
          <div className={"dashboard-sidebar"}>
              <div className={"dashboard-pfp-name"}>
                  <img src={user.google.picture} alt="Profile picture" className={"dashboard-pfp"}/>
                  <p>{user.google.name}</p>
              </div>
              <p>My donations</p>
              <p>My subscriptions</p>
          </div>
          <div className={"dashboard-main"}>
              <div className={"dashboard-top-row"}>
                <div className={"dashboard-info-box"}>
                    <p>23 <br/> donation</p>
                </div>
                  <div className={"dashboard-info-box"}>
                      <p>5 <br/> subscriptions</p>
                  </div>
                  <div className={"dashboard-info-box"}>
                      <p>750 000<br/> donated</p>
                  </div>
              </div>
              <div className={"dashboard-row-2"}>
                  <h2>Donation over time</h2>
                  <div className={"chart-wrap"}>
                      <div className={"chart2"}>
                          <Chart2/>
                      </div>
                      <div className={"chart2-sidebar"}>
                          <select name="months" id="months">
                              <option value="january">January</option>
                              <option value="february">February</option>
                              <option value="march">March</option>
                              <option value="april">April</option>
                          </select>
                          <div className={"chart2-circle"}>
                              <p>192 351 <br/> Avg per month</p>
                          </div>
                      </div>
                  </div>
              </div>
              <div className={"dashboard-row-3"}>
                  <h2>Donasjoner</h2>
                  <h4>sammenlignet med lignende bedrifter</h4>
                  <div className={"chart-wrap"}>
                      <div className={"chart1"}>
                          <Chart1/>
                      </div>
                      <div className={"chart1-sidebar"}>
                          <div className={"chart1-circle"}>
                              <p className={"percent-txt"}>25%</p>
                              <p className={"chart1-circle-txt"}>Bak gjennomsnittet</p>
                          </div>
                          <div className={"chart1-textbox"}>
                              <p>Basert p√• firma <br/> sin omsetning</p>
                          </div>
                      </div>
                  </div>
              </div>
          </div>
      </div>
  );
}