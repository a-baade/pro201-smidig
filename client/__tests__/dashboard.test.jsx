import ReactDOM from "react-dom";
import React, { createElement } from "react";
import { MemoryRouter } from "react-router-dom";
import { act } from "react-dom/test-utils";
import { Dashboard } from "../pages/dashboard";

describe("Dashboard component", () => {
  it("should test div element", async function () {
    const user = { google: { name: "test name", email: "test email" } };
    const divEl = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Dashboard user={user} />
      </MemoryRouter>,
      divEl
    );
    expect(divEl.innerHTML).toMatchSnapshot();
  });
  it("should show demo dashboard", async function () {
    const user = {
      google: {
        name: "test name",
      },
    };

    const element = document.createElement("div");
    await act(async () => {
      ReactDOM.render(
        <MemoryRouter>
          <Dashboard user={user} />
        </MemoryRouter>,
        element
      );
    });

    expect(
      Array.from(element.querySelectorAll("p")).map((e) => e.innerHTML)
    ).toEqual([
      "test name",
      "23 <br> donation",
      "5 <br> subscriptions",
      "750 000<br> donated",
      "192 351 <br> Avg per month",
      "25%",
      "Bak gjennomsnittet",
      "Basert på firma <br> sin omsetning",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
      "Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.",
    ]);
    expect.stringContaining("Profile for test name (test email)");
    expect(element.innerHTML).toMatchInlineSnapshot(
      `"<div class=\\"dashboard-page\\"><div class=\\"dashboard-sidebar\\"><div class=\\"dashboard-pfp-name\\"><img alt=\\"Profile picture\\" class=\\"dashboard-pfp\\"><p>test name</p></div><a href=\\"#subscriptions-a\\" class=\\"dash-link\\">My Subscriptions</a><a href=\\"#donations-a\\" class=\\"dash-link\\">My Donations</a></div><div class=\\"dashboard-main\\"><div class=\\"dashboard-top-row\\"><div class=\\"dashboard-info-box\\"><p>23 <br> donation</p></div><div class=\\"dashboard-info-box\\"><p>5 <br> subscriptions</p></div><div class=\\"dashboard-info-box\\"><p>750 000<br> donated</p></div></div><div class=\\"dashboard-row-2\\"><h2>Donation over time</h2><div class=\\"chart-wrap\\"><div class=\\"chart2\\"></div><div class=\\"chart2-sidebar\\"><select name=\\"months\\" id=\\"months\\"><option value=\\"january\\">January</option><option value=\\"february\\">February</option><option value=\\"march\\">March</option><option value=\\"april\\">April</option></select><div class=\\"chart2-circle\\"><p>192 351 <br> Avg per month</p></div></div></div></div><div class=\\"dashboard-row-3\\"><h2>Donasjoner</h2><h4>sammenlignet med lignende bedrifter</h4><div class=\\"chart-wrap\\"><div class=\\"chart1\\"></div><div class=\\"chart1-sidebar\\"><div class=\\"chart1-circle\\"><p class=\\"percent-txt\\">25%</p><p class=\\"chart1-circle-txt\\">Bak gjennomsnittet</p></div><div class=\\"chart1-textbox\\"><p>Basert på firma <br> sin omsetning</p></div></div></div></div><a id=\\"subscriptions-a\\"></a><div class=\\"dashboard-row-4\\"><h2>My subscriptions</h2><div class=\\"dash-carousel-wrapper\\"><div class=\\"carousel-container\\"><div class=\\"carousel-wrapper\\"><div class=\\"carousel-content-wrapper\\"><div class=\\"carousel-content show-3\\" style=\\"transform: translateX(-0%);\\"><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"cc-btn-wrap\\"><button class=\\"cc-btn\\">info</button><button class=\\"cc-btn\\">cancel</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"cc-btn-wrap\\"><button class=\\"cc-btn\\">info</button><button class=\\"cc-btn\\">cancel</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"cc-btn-wrap\\"><button class=\\"cc-btn\\">info</button><button class=\\"cc-btn\\">cancel</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"cc-btn-wrap\\"><button class=\\"cc-btn\\">info</button><button class=\\"cc-btn\\">cancel</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"cc-btn-wrap\\"><button class=\\"cc-btn\\">info</button><button class=\\"cc-btn\\">cancel</button></div></div></div></div></div></div><button class=\\"right-arrow\\">&gt;</button></div></div></div></div><a id=\\"donations-a\\"></a><div class=\\"dashboard-row-5\\"><h2>Previous donations</h2><div class=\\"dash-carousel-wrapper\\"><div class=\\"carousel-container\\"><div class=\\"carousel-wrapper\\"><div class=\\"carousel-content-wrapper\\"><div class=\\"carousel-content show-3\\" style=\\"transform: translateX(-0%);\\"><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"dc-btn-wrap\\"><button class=\\"cc-btn\\">info</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"dc-btn-wrap\\"><button class=\\"cc-btn\\">info</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"dc-btn-wrap\\"><button class=\\"cc-btn\\">info</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"dc-btn-wrap\\"><button class=\\"cc-btn\\">info</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"dc-btn-wrap\\"><button class=\\"cc-btn\\">info</button></div></div></div></div><div><div class=\\"card-wrap\\"><div class=\\"carousel-card\\"><div class=\\"card-header\\"></div><img src=\\"https://i.pinimg.com/originals/24/bb/38/24bb38268caffa8a18fdede24fd5dff9.png\\" alt=\\"placeholder\\" class=\\"pfp-placeholder\\"><p class=\\"carousel-card-text\\">Thank you for your generous gift to us! We are thrilled to have your support.<br> <br>You truly make the difference for us, and we are extremely grateful.</p><div class=\\"dc-btn-wrap\\"><button class=\\"cc-btn\\">info</button></div></div></div></div></div></div><button class=\\"right-arrow\\">&gt;</button></div></div></div></div></div></div>"`
    );
    expect(element.innerHTML).toMatchSnapshot();
  });
});
