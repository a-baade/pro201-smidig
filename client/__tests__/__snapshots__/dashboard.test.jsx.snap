// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should show demo dashboard 2`] = `"<div class=\\"dashboard-page\\"><div class=\\"dashboard-sidebar\\"><div class=\\"dashboard-pfp-name\\"><img alt=\\"Profile picture\\" class=\\"dashboard-pfp\\"><p>test name</p></div><p>My donations</p><p>My subscriptions</p></div><div class=\\"dashboard-main\\"><div class=\\"dashboard-top-row\\"><div class=\\"dashboard-info-box\\"><p>23 <br> donation</p></div><div class=\\"dashboard-info-box\\"><p>5 <br> subscriptions</p></div><div class=\\"dashboard-info-box\\"><p>750 000<br> donated</p></div></div><div class=\\"dashboard-row-2\\"><h2>Donation over time</h2><div class=\\"chart-wrap\\"><div class=\\"chart2\\"></div><div class=\\"chart2-sidebar\\"><select name=\\"months\\" id=\\"months\\"><option value=\\"january\\">January</option><option value=\\"february\\">February</option><option value=\\"march\\">March</option><option value=\\"april\\">April</option></select><div class=\\"chart2-circle\\"><p>192 351 <br> Avg per month</p></div></div></div></div><div class=\\"dashboard-row-3\\"><h2>Donasjoner</h2><h4>sammenlignet med lignende bedrifter</h4><div class=\\"chart-wrap\\"><div class=\\"chart1\\"></div><div class=\\"chart1-sidebar\\"><div class=\\"chart1-circle\\"><p class=\\"percent-txt\\">25%</p><p class=\\"chart1-circle-txt\\">Bak gjennomsnittet</p></div><div class=\\"chart1-textbox\\"><p>Basert på firma <br> sin omsetning</p></div></div></div></div></div></div>"`;

exports[`Dashboard component should test div element 1`] = `"<div class=\\"dashboard-page\\"><div class=\\"dashboard-sidebar\\"><div class=\\"dashboard-pfp-name\\"><img alt=\\"Profile picture\\" class=\\"dashboard-pfp\\"><p>test name</p></div><p>My donations</p><p>My subscriptions</p></div><div class=\\"dashboard-main\\"><div class=\\"dashboard-top-row\\"><div class=\\"dashboard-info-box\\"><p>23 <br> donation</p></div><div class=\\"dashboard-info-box\\"><p>5 <br> subscriptions</p></div><div class=\\"dashboard-info-box\\"><p>750 000<br> donated</p></div></div><div class=\\"dashboard-row-2\\"><h2>Donation over time</h2><div class=\\"chart-wrap\\"><div class=\\"chart2\\"></div><div class=\\"chart2-sidebar\\"><select name=\\"months\\" id=\\"months\\"><option value=\\"january\\">January</option><option value=\\"february\\">February</option><option value=\\"march\\">March</option><option value=\\"april\\">April</option></select><div class=\\"chart2-circle\\"><p>192 351 <br> Avg per month</p></div></div></div></div><div class=\\"dashboard-row-3\\"><h2>Donasjoner</h2><h4>sammenlignet med lignende bedrifter</h4><div class=\\"chart-wrap\\"><div class=\\"chart1\\"></div><div class=\\"chart1-sidebar\\"><div class=\\"chart1-circle\\"><p class=\\"percent-txt\\">25%</p><p class=\\"chart1-circle-txt\\">Bak gjennomsnittet</p></div><div class=\\"chart1-textbox\\"><p>Basert på firma <br> sin omsetning</p></div></div></div></div></div></div>"`;
